next-docs/pages/guides/hello-world/add-system.mdx

```diff
@@ -0,0 +1,296 @@
+          console.log("new counter value:", await decrement());
+        };
+
+        // https://vitejs.dev/guide/env-and-mode.html
+        if (import.meta.env.DEV) {
+          const { mount: mountDevTools } = await import("@latticexyz/dev-tools");
+          mountDevTools({
+            config: mudConfig,
+            publicClient: network.publicClient,
+            walletClient: network.walletClient,
+            latestBlock$: network.latestBlock$,
+            storedBlockLogs$: network.storedBlockLogs$,
+            worldAddress: network.worldContract.address,
+            worldAbi: network.worldContract.abi,
+            write$: network.write$,
+            recsWorld: network.world,
+          });
+        }
+        ```
+
+    </CollapseCode>
+
+           <details>
+
+    <summary>Explanation</summary>
+
+        {" "}
+        <p></p>
+
+        ```typescript
+        const {
+          components,
+          systemCalls: { decrement, increment },
+        } = await setup();
+        ```
+
+        This syntax means the we call [`setup()`](https://github.com/latticexyz/mud/blob/main/examples/minimal/packages/client-vanilla/src/mud/setup.ts), and then set `components`, `systemCalls.increment`, and `systemCalls.decrement` to the values provided in the hash returned by this function.
+        `systemCalls` comes from `createSystemCalls()`, which we modified in the previous step.
+
+        ```typescript
+        (window as any).decrement = async () => {
+          console.log("new counter value:", await decrement());
+        };
+        ```
+
+        We need to make `decrement` available to our application code.
+        Most frameworks have a standard mechanism to do this, but we are using `vanilla`, which doesn't - so we add it to `window` which is a global variable.
+
+           </details>
+
+1.  Modify `packages/client/index.html` to add a decrement button.
+    This is the file after the changes:
+
+    <CollapseCode>
+
+        ```html filename="index.html" copy showLineNumbers {12}
+        <!DOCTYPE html>
+        <html lang="en">
+          <head>
+            <meta charset="UTF-8" />
+            <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+            <title>a minimal MUD client</title>
+          </head>
+          <body>
+            <script type="module" src="/src/index.ts"></script>
+            <div>Counter: <span id="counter">0</span></div>
+            <button onclick="window.increment()">Increment</button>
+            <button onclick="window.decrement()">Decrement</button>
+          </body>
+        </html>
+        ```
+
+    </CollapseCode>
+
+           <details>
+
+    <summary>Explanation</summary>
+
+        {" "}
+        <p></p>
+
+        ```html
+        <button onclick="window.decrement()">Decrement</button>
+        ```
+
+        Create a [`button`](https://www.w3schools.com/tags/tag_button.asp) with an [`onClick`](https://www.w3schools.com/tags/ev_onclick.asp) property.
```

Comment:
```suggestion
        Create a [`button`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button) with an [`onClick`](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events) property.
```

Reply:
btw, after looking at https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events i am realizing we should probably not use the `window` and inline `onclick` approach but rather give the button an ID and then do this in the javascript code:

```ts
const incrementButton = document.querySelector("#increment-button");
incrementButton.addEventListener("click", increment);
```

One could argue that it doesn't matter much because it's unlikely folks would _actually_ use plain html/js (besides learning purposes, or for integrating in a separate framework), but I think it could give a bad first impression if we're using bad practices in the vanilla template.

Reply:
Merge https://github.com/latticexyz/mud/pull/1774 and I'll fix this here too.